+ adding new uProc integration

mcolomer opened the pull request:
Hi,

This is Miquel from uProc.

We are providing PR for uProc integration into n8n.

Currently existing groups, tools and parameters are loaded in real time at GroupDescription.ts and ToolDescription.ts files using existing json dumped from our system. 

Data is loaded when running n8n and converted to n8n formatted arrays to load selectors properly.
We are interested to load new tools getting jsons from our platform instead of pushing them regularly to the repository.

Could you verify if this can be fixed someway to avoid pushing code with new tools manually?

Thanks!

mcolomer committed:
+ adding new uProc integration

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/n8n-io/n8n?pullRequest=795) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/n8n-io/n8n?pullRequest=795) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/n8n-io/n8n?pullRequest=795) it.</sub>

mcolomer committed:
+ capitalize tool name
+ fix parameter display name
+ add placeholders for parameter
+ add tool description
+ optimize generated .ts for groups and tools

mcolomer commented:
Hi Ricardo,

Thank you for reviewing the integration.
We have reviewed all the code and applied all your recommended changes
(some of them are hard - like parameter description - because it needs to
review all existing tools - more than 400 -).

Could you review last commit if matches your requirements?

Thank you!

Missatge de Ricardo Espinoza <notifications@github.com> del dia ds., 1
d’ag. 2020 a les 2:03:

> @mcolomer <https://github.com/mcolomer> just reviewed the integration,
> and it was not working, it took me quite some time to get it working. Maybe
> you missed pushing a commit? These are some of the things that have to be
> fixed.
>
>    - Parameters name labels have to start with Uppercase
>
> [image: image]
> <https://user-images.githubusercontent.com/16496553/89088600-4cb14680-d367-11ea-8fcf-1780b0625ccd.png>
>
>    - Same with tools names, there are inconsistencies, some of them use
>    uppercase on each word and some of them use a mix of both.
>
> [image: image]
> <https://user-images.githubusercontent.com/16496553/89088589-415e1b00-d367-11ea-9fa6-2f3e99197cd3.png>
>
>    -
>
>    Parameter descriptions are not clear and seem like all of them are
>    just type string and do not take advantage of all n8n UI component types.
>    -
>
>    The processor JSON is HUGE, and it seems like there is a lot of
>    unnecessary data there. Can that be cleaned just to hold the necessary data
>    to work?
>
> Let me know if something it's not clear.
>
> I pushed the changes I made to this branch so you can start from there
> RicardoE105@2c35b9e
> <https://github.com/RicardoE105/n8n/commit/2c35b9e0f55add00492d16af8aac208a57db08a0>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/n8n-io/n8n/pull/795#issuecomment-667432478>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AENWRJQRB35TW3WTANHBCL3R6NLWTANCNFSM4PG4MT6A>
> .
>


-- 
Miquel Colomer
mcolomer@gmail.com

mcolomer committed:
+ remove old functions

mcolomer committed:
+ added new tools

mcolomer committed:
+ add new tools + fix Info on tool

mcolomer commented:
Hi @RicardoE105,

We just pushed new changes that add some additional info for our tools.
Could you check it please?

Thanks!

mcolomer committed:
+ fixed Info

RicardoE105 commented:
@mcolomer Sorry for the late response. Will review again tomorrow and will keep you posted. Thanks.

RicardoE105 commented:
Hey, @mcolomer again sorry for the late response. Checking this right now. Quick question, how do you test the node if the node is not added to the package.json file? did you forget to push a change?

mcolomer committed:
+ fix tools description

mcolomer committed:
+ add UProc node to package.json

mcolomer commented:
Hi @RicardoE105 ,

Thanks for your reply.
I have added UProc entries (credentials and node) to package.json and pushed.

Currently, I was testing two ways:
1. Adding UProc node to ~/.n8n/custom directory and running docker container.
2. Creating npm package locally, linking with another n8n custom installation and running binary at node_modules/n8n/bin/n8n. 

Package.json declaration was not required to load node properly in both cases.

Let me know if this matches your requirements.

mcolomer committed:
+ added new tools

mcolomer commented:
We have uploaded new tools to tools.ts.
Related to this PR, is there any estimated date to add to the main tree?

Thanks!

RicardoE105 commented:
Hey guys, very sorry that I have not been able to get to this yet. I will provide feedback this week. Just move it to the top of my todo list. Thanks.

mcolomer committed:
+ added new tools

mcolomer committed:
+ fix tool description

mcolomer committed:
+ fix CamelCase subtitle
+ fix node displayName
+ add new tools

mcolomer committed:
+ fix info link

mcolomer committed:
+ add start, data and end callbacks to decouple response

RicardoE105 commented:
@mcolomer quick update here guys, the blocker stopping us from merging this PR has been removed. I'm gonna review again this week to make sure everything is still ok, and then we can move forward. Sorry for the late response, and thanks.

mcolomer committed:
+ added new tools

mcolomer commented:
Great news @RicardoE105.

We have pushed several changes to #954 with last tools added to system.
Could you check it please?

Thanks!

mcolomer committed:
+ change display name on selected

mcolomer committed:
+ added new tools

mcolomer committed:
+ added new tools

mcolomer committed:
+ added new tools

mcolomer committed:
+ added new tools

mcolomer committed:
+ remove old folder + fix param

mcolomer committed:
+ added new tools

janober commented:
@mcolomer thanks a lot. Was just doing my final review. Fixed some things (mainly formatting) and pushed to branch
https://github.com/n8n-io/n8n/tree/mcolomer-uProc_integration

Wanted to merge but sadly does the most functionality not seem to work for me. Get a lot of "result: true" and "Sample Text" everywhere but almost never the actual results. What is the reason for that? The account or the API?
![Screenshot from 2020-12-19 17-16-20](https://user-images.githubusercontent.com/6249596/102693910-20b33680-421e-11eb-9b9e-96feab561746.png)

mcolomer commented:
Hi Jan,

This is related with the api key used.
Basically there two types at app.uproc.io/#/settings/integration :

1) Api Key (test) that returns faked data inside message property (the one
you are using).
2) API key (real) that returns real data for all your queries.

Let me know if changing to API Key real does the trick.

Regards,

On Sat, Dec 19, 2020, 17:19 Jan <notifications@github.com> wrote:

> @mcolomer <https://github.com/mcolomer> thanks a lot. Was just doing my
> final review. Fixed some things (mainly formatting) and pushed to branch
> https://github.com/n8n-io/n8n/tree/mcolomer-uProc_integration
>
> Wanted to merge but sadly does the most functionality not seem to work for
> me. Get a lot of "result: true" and "Sample Text" everywhere but almost
> never the actual results. What is the reason for that? The account or the
> API?
> [image: Screenshot from 2020-12-19 17-16-20]
> <https://user-images.githubusercontent.com/6249596/102693910-20b33680-421e-11eb-9b9e-96feab561746.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/n8n-io/n8n/pull/795#issuecomment-748494071>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AENWRJRFN5A4VJONHRAXFPDSVTHCBANCNFSM4PG4MT6A>
> .
>

janober commented:
Thanks a lot. That explains it. Just got merged with #1263

mcolomer commented:
Thanks a lot Jan!

On Sat, Dec 19, 2020, 18:24 Jan <notifications@github.com> wrote:

> Thanks a lot. That explains it. Just got merged with #1263
> <https://github.com/n8n-io/n8n/pull/1263>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/n8n-io/n8n/pull/795#issuecomment-748501207>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AENWRJSP27YVM3GFHLRWXPTSVTOUHANCNFSM4PG4MT6A>
> .
>

janober commented:
Got released with n8n@0.99.0
